packages/x-components/tests/e2e/cucumber/next-queries/next-queries.spec.ts

```diff
@@ -18,14 +20,22 @@ Given(
         }
       }
     };
-    cy.visit('/test/next-queries', {
-      qs: {
-        xModules: JSON.stringify(config)
-      }
-    });
   }
 );
 
+And('next queries response being mock {string}', (mockName: string) => {
+  cy.intercept('https://api.empathy.co/nextQueries', async req => {
+    const module = await import('./stubs/next-queries.stub');
+    req.reply(module[mockName as keyof typeof module]);
+  }).as('interceptedNextQueries');
+
+  cy.visit('/test/next-queries?useMockedAdapter=true', {
```

Comment:
So every time you try to mock the next queries response it reloads the page?  Does that make sense? I think those should be separate steps.

Can't we intercept the nqs before loading the page if we have them in separate steps?

Reply:
I think I understand now why you said that you needed to mock the search API. I would propose an alternative approach, which is extending the `EmpathyAdapter` until we have the full mocked adapter.

Something like this:

```
class E2ETestsAdapter extends EmpathyAdapter {
  getNextQueries(request: NextQueriesRequest): Promise<NextQueriesResponse> {
    return mockFetch(request, 'getNextQueries');
  }
  getTopRecommendations(request: TopRecommendationsRequest): Promise<TopRecommendationsResponse> {
    return mockFetch(request, 'getTopRecommendations');
  }
}

export const mockedAdapter = new EmpathyAdapterBuilder(
  undefined,
  undefined,
  E2ETestsAdapter
).build();
```

Then only the overridden methods are mocked, the remaining ones can continue with its standard flow. We can also remove the `useMockedAdapter` parameter from the URL and its related code.